src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1525,7 +1542,7 @@ public function toArray(): array
             throw new JsonException(sprintf('JSON content was expected to decode to an array, "%s" returned.', get_debug_type($content)));
         }
 
-        return $content;
+        return $this->arrayContent = $content;
```

Comment:
Looks for me like a bc break as, it not longer allows to manipulate the content payload inside of a request listener.

Reply:
how do you do that currently?

Reply:
There is no `setContent` allowing to modify it in listeners

Reply:
I see. Did recheck it the current manipulation I know is on the `$request->request->set('...')` not the content.

Reply:
Thinking twice about this, it's still possible to update the content property since it's protected.
I've updated the PR to remove this memoization. Let's consider it separately from this PR.